src/libutil/file-system.hh

```diff
@@ -194,7 +190,7 @@ void moveFile(const Path & src, const Path & dst);
  * with the guaranty that the destination will be “fresh”, with no stale inode
  * or file descriptor pointing to it).
  */
-void copyFile(const Path & oldPath, const Path & newPath, bool andDelete);
+void copyFile(const std::filesystem::directory_entry & from, const std::filesystem::path & to, bool andDelete);
```

Comment:
Why is `from` a `directory_entry` rather than a `path`?

Reply:
I have changed it to use `path` instead in https://github.com/NixOS/nix/pull/10685/commits/62e1ea2f4b563d73fac8d48feae0e9968c9c5bc9. Not sure why it was `directory_entry` before.